src/neo/SmartContract/Native/Tokens/NeoToken.cs

```diff
@@ -50,45 +55,28 @@ protected override void OnBalanceChanging(ApplicationEngine engine, UInt160 acco
 
         private void DistributeGas(ApplicationEngine engine, UInt160 account, NeoAccountState state)
         {
-            BigInteger gas = CalculateBonus(state.Balance, state.BalanceHeight, engine.Snapshot.PersistingBlock.Index);
+            BigInteger gas = CalculateBonus(engine.Snapshot, state.Balance, state.BalanceHeight, engine.Snapshot.PersistingBlock.Index);
             state.BalanceHeight = engine.Snapshot.PersistingBlock.Index;
             GAS.Mint(engine, account, gas);
         }
 
-        private BigInteger CalculateBonus(BigInteger value, uint start, uint end)
+        private BigInteger CalculateBonus(StoreView snapshot, BigInteger value, uint start, uint end)
         {
             if (value.IsZero || start >= end) return BigInteger.Zero;
             if (value.Sign < 0) throw new ArgumentOutOfRangeException(nameof(value));
-            BigInteger amount = BigInteger.Zero;
-            uint ustart = start / Blockchain.DecrementInterval;
-            if (ustart < Blockchain.GenerationAmount.Length)
-            {
-                uint istart = start % Blockchain.DecrementInterval;
-                uint uend = end / Blockchain.DecrementInterval;
-                uint iend = end % Blockchain.DecrementInterval;
-                if (uend >= Blockchain.GenerationAmount.Length)
-                {
-                    uend = (uint)Blockchain.GenerationAmount.Length;
-                    iend = 0;
-                }
-                if (iend == 0)
-                {
-                    uend--;
-                    iend = Blockchain.DecrementInterval;
-                }
-                while (ustart < uend)
-                {
-                    amount += (Blockchain.DecrementInterval - istart) * Blockchain.GenerationAmount[ustart];
-                    ustart++;
-                    istart = 0;
-                }
-                amount += (iend - istart) * Blockchain.GenerationAmount[ustart];
-            }
-            return value * amount * GAS.Factor / TotalAmount;
+            var gasPerBlock = GetGasPerBlock(snapshot);
+            return value * gasPerBlock * (end - start) * NeoHolderRewardRatio / 100 / TotalAmount;
```

Comment:
The value of `GasPerBlock` is read from storage. What if the value changed?

Reply:
If we want to allow `GetGasPerBlock` to be modified, we need to record the historical value of `GasPerBlock`, or the incremental value. If so, we'd better add `SetGasPerBlock` method.

Reply:
The committee has the permission to invoke `SetGasPerBlock`. And there is a range restriction on the value from `0` to `10` GAS pre block.

Reply:
Ok, I'll add it.

Reply:
Done.